netket/_scipy_integrators.py

```diff
@@ -0,0 +1,131 @@
+import scipy
+import numpy as np
+from scipy.integrate import OdeSolver, DenseOutput
+from warnings import warn
+
+
+def validate_first_step(first_step, t0, t_bound):
+    """Assert that first_step is valid and return it."""
+    if first_step <= 0:
+        raise ValueError("`first_step` must be positive.")
+    if first_step > np.abs(t_bound - t0):
+        raise ValueError("`first_step` exceeds bounds.")
+    return first_step
+
+
+def warn_extraneous(extraneous):
+    """Display a warning for extraneous keyword arguments.
+
+    The initializer of each solver class is expected to collect keyword
+    arguments that it doesn't understand and warn about them. This function
+    prints a warning for each key in the supplied dictionary.
+
+    Parameters
+    ----------
+    extraneous : dict
+        Extraneous keyword arguments
+    """
+    if extraneous:
+        warn(
+            "The following arguments have no effect for a chosen solver: {}.".format(
+                ", ".join("`{}`".format(x) for x in extraneous)
+            )
+        )
+
+
+class EulerSolver(OdeSolver):
+    """Base class for explicit Runge-Kutta methods."""
+
+    B = np.array(
+        [
+            1,
+        ]
+    )
+    order = NotImplemented
+    error_estimator_order = NotImplemented
+    n_stages = 1
+
+    def __init__(
+        self,
+        fun,
+        t0,
+        y0,
+        t_bound,
+        step=None,
+        vectorized=False,
+        first_step=None,
+        **extraneous,
+    ):
+        warn_extraneous(extraneous)
+        super(EulerSolver, self).__init__(
+            fun, t0, y0, t_bound, vectorized, support_complex=True
+        )
```

Comment:
```suggestion
        super().__init__(
            fun, t0, y0, t_bound, vectorized, support_complex=True
        )
```
In Python 3 `super(Class, self)` is equivalent to `super()` (in a method of `Class`).